From 79a4d199f377f757880cc3f38d89f6cbe8ae2258 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 4 Jul 2019 22:12:10 -0700
Subject: [PATCH 0641/1852] v4l: xilinx: sdirxss: Call
 xsdirx_streamflow_control() directly

commit 7f8abf54428b034413f663a86f324d009f3141bc from
https://github.com/Xilinx/linux-xlnx.git

Remove unnecessary wrapper functions of downstreamcb(), start_stream()
and stop_stream() and directly call xsdirx_streamflow_control() with
correct flag directly.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../media/platform/xilinx/xilinx-sdirxss.c    | 21 +++----------------
 1 file changed, 3 insertions(+), 18 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index 963d446c226f..d6f0b1da6482 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -673,11 +673,6 @@ static void xsdirx_streamflow_control(struct xsdirxss_core *core, bool enable)
 	}
 }
 
-static void xsdirx_streamdowncb(struct xsdirxss_core *core)
-{
-	xsdirx_streamflow_control(core, false);
-}
-
 static void xsdirxss_get_framerate(struct v4l2_fract *frame_interval,
 				   u32 framerate)
 {
@@ -1010,7 +1005,7 @@ static irqreturn_t xsdirxss_irq_handler(int irq, void *dev_id)
 
 		dev_dbg(core->dev, "video lock/unlock interrupt\n");
 
-		xsdirx_streamdowncb(core);
+		xsdirx_streamflow_control(core, false);
 		state->streaming = false;
 
 		val1 = xsdirxss_read(core, XSDIRX_MODE_DET_STAT_REG);
@@ -1313,16 +1308,6 @@ static int xsdirxss_log_status(struct v4l2_subdev *sd)
 	return 0;
 }
 
-static void xsdirxss_start_stream(struct xsdirxss_state *xsdirxss)
-{
-	xsdirx_streamflow_control(&xsdirxss->core, true);
-}
-
-static void xsdirxss_stop_stream(struct xsdirxss_state *xsdirxss)
-{
-	xsdirx_streamflow_control(&xsdirxss->core, false);
-}
-
 /**
  * xsdirxss_g_frame_interval - Get the frame interval
  * @sd: V4L2 Sub device
@@ -1379,7 +1364,7 @@ static int xsdirxss_s_stream(struct v4l2_subdev *sd, int enable)
 			return -EINVAL;
 		}
 
-		xsdirxss_start_stream(xsdirxss);
+		xsdirx_streamflow_control(core, true);
 		xsdirxss->streaming = true;
 		dev_dbg(core->dev, "Streaming started\n");
 	} else {
@@ -1388,7 +1373,7 @@ static int xsdirxss_s_stream(struct v4l2_subdev *sd, int enable)
 			return -EINVAL;
 		}
 
-		xsdirxss_stop_stream(xsdirxss);
+		xsdirx_streamflow_control(core, false);
 		xsdirxss->streaming = false;
 		dev_dbg(core->dev, "Streaming stopped\n");
 	}
-- 
2.31.1

